### PR TITLE
Update modal design

### PR DESCRIPTION
**Pull request by spzm**
# What does this PR do:
  
     - The design of modal was updated regarding issue #314.
     - Introduce small and big (by default) states of modal regarding design.
     - ModalContent component introduced to unify content body inside modal window.

# Where should the reviewer start:

    - Test modal window with styleguide.
    - Check backward compatibility according current usage of modal.

### CODE REVIEW

File: components/index.scss [hunk 1]
```diff
@@ -17,6 +17,7 @@
 @import "./loadingOverlay/loadingOverlay.css";
 @import "./messageBox/messageBox.scss";
 @import "./modal/modal.scss";
+@import "./modal/modalContent.scss";
```
Comment (AlleeX) [inline components/index.scss]: only css! ;)

File: components/modal/modalContent.js [hunk 2]
```diff
@@ -0,0 +1,39 @@
+import PropTypes from 'prop-types';
+import React from 'react';
+
+
```
Comment (AlleeX) [inline components/modal/modalContent.js]: extra line

File: themes/_default.yaml [hunk 3]
```diff
@@ -434,6 +434,10 @@ modal:
       color: "var(--tx-generic-color-active)"
     icon:
       color: "var(--tx-generic-color-secondary-darker)"
+  modal-content:
+    background:
+      color: "var(--tx-generic-color-blank)"
+    border-radius: 8px;
```
Comment (AlleeX) [inline themes/_default.yaml]: are we really have dependency from other themes?

**Comment by EduardTrutsyk** [comment]
why the modal closing without any animation?

File: components/modal/modal.js [hunk 4]
```diff
@@ -256,6 +258,10 @@ Modal.propTypes = {
    * Determine whether a overlay is visible or not
    */
   overlay: PropTypes.bool,
+  /**
+   * Price's size
```
Comment (iwwwi) [inline components/modal/modal.js]: Modal size

**Comment by spzm** [comment]
In master I see same close-without-animation behavior. Also, the last modal has close animation. I'm going to check is it possible to fix with this PR.

File: components/modal/modalContent.css [hunk 5]
```diff
@@ -0,0 +1,38 @@
+.ui-modal-content-block {
+  background-color: var(--tx-generic-color-blank);
+  border-radius: 8px;
+}
+
+.ui-modal-content-block__title {
+  font-weight: 600;
```
Comment (maximuk) [inline components/modal/modalContent.css]: please sort and use variable for `font-weight` (also in other places)

File: components/modal/modalContent.css [hunk 6]
```diff
@@ -0,0 +1,38 @@
...
+  border-bottom: 2px solid var(--tx-modal-content-background-color);
+  padding: 24px;
+}
+
+.ui-modal-content-block__container {
+  border-radius: 8px;
```
Comment (maximuk) [inline components/modal/modalContent.css]: we have variable for `border-radius` - `--tx-generic-border-radius` (also should be fixed in other places)

File: components/modal/modalContent.js [hunk 7]
```diff
@@ -0,0 +1,38 @@
...
+ModalContent.propTypes = {
+  /**
+   * The modal dialog's body
+   */
+  children: PropTypes.node,
+  /**
+   * Title text of content
+   */
+  title: PropTypes.node,
```
Comment (maximuk) [inline components/modal/modalContent.js]: please sort

File: components/selectionBlock/selectionBlock.css [hunk 8]
```diff
@@ -66,6 +66,7 @@
   justify-content: center;
   margin: 0;
   width: 290px;
+  min-width: 290px;
```
Comment (maximuk) [inline components/selectionBlock/selectionBlock.css]: please sort

File: components/modal/modalContent.js [hunk 9]
```diff
@@ -0,0 +1,38 @@
...
+  /**
+   * Set of custom modifications.
+   */
+  mods: PropTypes.arrayOf(PropTypes.string),
```
Comment (iwwwi) [inline components/modal/modalContent.js]: pls do not add `mods` because those are planned to be deprecated.. better do add `classname` if is needed at all... (i do not see where you use the `mods` in this component

File: components/modal/modal.js [hunk 10]
```diff
@@ -256,6 +258,10 @@ Modal.propTypes = {
    * Determine whether a overlay is visible or not
    */
   overlay: PropTypes.bool,
+  /**
+   * Modal size
+   */
+  size: PropTypes.oneOf(['small', 'big']),
```
Comment (iwwwi) [inline components/modal/modal.js]: Since we have only 2 possible choices is it better to go with `true/false` variant? In that case we won't have unused class like - `size_big`

**Comment by EduardTrutsyk** [comment]
maybe make sense to add `className` for `Modal` and for `ModalContent`?

File: components/modal/modal.js [hunk 11]
```diff
@@ -200,6 +204,7 @@ Modal.defaultProps = {
   onClose: null,
   onOverlayClick: null,
   overlay: true,
+  size: false,
```
Comment (froskie) [inline components/modal/modal.js]: Passing `false` is odd here, isn't it? Wouldn't `null` be a better default?

Comment (froskie) [reply components/modal/modal.js]: Also, where is this being used?

Comment (spzm) [reply components/modal/modal.js]: There are two sizes of modal available: big (max-width: 780), small (max-width: 480).
If I specify size as Boolean I prefer set it as *false* instead of *null* value.

Comment (iwwwi) [reply components/modal/modal.js]: I guess here the prop should be `isSmall` and then will have the default `false`

File: components/modal/modal.scss [hunk 12]
```diff
@@ -114,31 +113,48 @@
 }
 
 .ui-modal__header {
+  align-items: center;
   border-bottom: 2px solid var(--tx-modal-content-devider-color);
   color: var(--tx-modal-color-dark);
+  display: flex;
+
```
Comment (iwwwi) [inline components/modal/modal.scss]: remove empty line pls

File: components/modal/modal.js (hunk 11 above)
Comment (spzm) [reply components/modal/modal.js]: yep, thanks, fixing

File: components/selectionBlock/selectionBlock.css [hunk 13]
```diff
@@ -65,6 +65,7 @@
   display: flex;
   justify-content: center;
   margin: 0;
+  min-width: 290px;
```
Comment (asci) [inline components/selectionBlock/selectionBlock.css]: why is this required?

Comment (spzm) [reply components/selectionBlock/selectionBlock.css]: Ah, forgot to mention in description, small fix for selection block.
